templates/macros.html

```diff
@@ -0,0 +1,122 @@
+    * `limits` A non-null `Limits` struct
+#}
+{% macro crate_limits(limits) %}
+    <table class="pure-table pure-table-horizontal">
+        <tbody>
+            <tr>
+                <td>Available RAM</td>
+                <td>{{ limits.memory | filesizeformat }}</td>
+            </tr>
+
+            <tr>
+                <td>Maximum rustdoc execution time</td>
+                <td>{{ limits.timeout.secs | timeformat }}</td>
+            </tr>
+
+            <tr>
+                <td>Maximum size of a build log</td>
+                <td>{{ limits.max_log_size | filesizeformat }}</td>
+            </tr>
+
+            <tr>
+                <td>Network access</td>
+                <td>
+                    {%- if limits.networking -%}
+                        allowed
+                    {%- else -%}
+                        blocked
+                    {%- endif -%}
+                </td>
+            </tr>
+
+            <tr>
+                <td>Maximum number of build targets</td>
+                <td>{{ limits.targets }}</td>
+            </tr>
+        </tbody>
+    </table>
+{% endmacro crate_limits %}
+
+{# Constructs a title based on the given crate name and version #}
+{% macro doc_title(name, version) %}
+    {%- if name -%}
+        {{ name }} {{ version | default(value="") }} - Docs.rs
+    {%- else -%}
+        Docs.rs
+    {%- endif -%}
+{% endmacro doc_title %}
+
+{#
+    Constructs a list of a crate's releases
+    * `name` The crate's name as a string
+    * `releases` A list of crate releases where each release has the following fields:
+        * `version` A string of the release's version
+        * `yanked` A boolean of the release's yanked status
+        * `build_status` A boolean of the crate's build status (true for built, false for failed build)
+#}
+{% macro releases_list(name, releases) %}
+    {%- for release in releases -%}
+        {# The url for the release, `/crate/:name/:version` #}
+        {%- set release_url = "/crate/" ~ name ~ "/" ~ release.version -%}
+        {# The release's name and version, `:name-:version` #}
+        {%- set release_name = name ~ "-" ~ release.version -%}
+
+        <li class="pure-menu-item">
+            {# If the release has been yanked and failed to build, display a warning #}
+            {%- if release.yanked and release.build_status -%}
+                <a href="{{ release_url }}" class="pure-menu-link warn" title="{{ release_name }} is yanked">
+                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                </a>
+
+            {# If the release is yanked but built, display an warning #}
+            {%- elif release.yanked and not release.build_status -%}
+                <a href="{{ release_url }}" class="pure-menu-link warn"
+                    title="{{ release_name }} is yanked and docs.rs failed to build it">
+                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                </a>
```

Comment:
The comments here do not match the code.
```suggestion
            {# If the release is yanked but built, display an warning #}
            {%- if release.yanked and release.build_status -%}
                <a href="{{ release_url }}" class="pure-menu-link warn" title="{{ release_name }} is yanked">
                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
                </a>

            {# If the release has been yanked and failed to build, display a warning #}
            {%- elif release.yanked and not release.build_status -%}
                <a href="{{ release_url }}" class="pure-menu-link warn"
                    title="{{ release_name }} is yanked and docs.rs failed to build it">
                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
                </a>
```